bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/rest/JettyAdminServiceV6.java

```diff
@@ -87,12 +87,16 @@ private String updateSid() throws PiHoleException {
         var request = client.newRequest(authURI).timeout(TIMEOUT_SECONDS, SECONDS).method(HttpMethod.POST)
                 .header(HttpHeader.ACCEPT, "application/json").content(new StringContentProvider(tokenJson));
         var content = send(request).getContentAsString();
+        logger.debug("Session update request answer: {}", content);
 
         if (gson.fromJson(content, SessionAnswer.class) instanceof SessionAnswer answer) {
-            Session session = answer.session();
-            sid = session.sid();
-            sessionValidity = Instant.now().plusSeconds(session.cautiousValidity());
-            return session.sid();
+            if (answer.session() instanceof Session session) {
+                sid = session.sid();
+                sessionValidity = Instant.now().plusSeconds(session.cautiousValidity());
+                return session.sid();
+            } else {
+                throw new PiHoleException("Received an empty session");
+            }
```

Comment:
Oh come one... Instead of using smart instance of just write a simple null-check. Nobody will be thinking what type `session()` returns that we need to check it.

```suggestion
            Session session = answer.session();
            If(session == null){
                 throw new PiHoleException("Received an empty session");
            }
            sid = session.sid();
            sessionValidity = Instant.now().plusSeconds(session.cautiousValidity());
            return session.sid();
```